### PR TITLE
Fixes #2592 - No indication when waiting for host dashboard button action...

### PR DESCRIPTION
**Pull request by boaz0**
... to complete

### CODE REVIEW

**Comment by boaz0** [comment]
- @shlomizadok @tbrisker @domcleal @eLobato , do you have any suggestions/ideas regarding my implementation?

**Comment by lzap** [comment]
Hehe, let me rephrase:

Must be in the format Fixes/refs #redmine_number - **brief** description

File: app/assets/javascripts/hosts.js [hunk 1]
```diff
@@ -22,4 +22,17 @@ $(function () {
...
+
+  var action_buttons = $(".btn-toolbar a");
+  var wait_msg = $('#processing_message');
+  $(action_buttons).on('click', function () {
+    $(action_buttons).prop('disabled',true);
+    $(this).prop('disabled', false);
+    if ($(this).attr('id') !== 'build-review') {
+      if ($(this).attr('id') === 'delete-button' && !confirm("Are you sure?"))
+        return false;
+      $(wait_msg).modal('show');
```
Comment (tbrisker) [inline app/assets/javascripts/hosts.js]: `action_buttons` and `wait_msg` are already jQuery objects, no need to wrap them in $() again

File: app/assets/javascripts/hosts.js [hunk 2]
```diff
@@ -22,4 +22,17 @@ $(function () {
...
     $("#build-review").click();
   });
+
+  var action_buttons = $(".btn-toolbar a");
+  var wait_msg = $('#processing_message');
+  $(action_buttons).on('click', function () {
+    $(action_buttons).prop('disabled',true);
+    $(this).prop('disabled', false);
+    if ($(this).attr('id') !== 'build-review') {
+      if ($(this).attr('id') === 'delete-button' && !confirm("Are you sure?"))
```
Comment (ohadlevy) [inline app/assets/javascripts/hosts.js]: note that confirm string is i18n extracted

**Comment by boaz0** [comment]
thanks @ohadlevy @tbrisker & @lzap for your comments.

**Comment by dLobatog** [comment]
@ripcurld00d How could I test this? The build button works very differently than it did when #2592 was opened.

**Comment by boaz0** [comment]
### goal

the motivation behind this pr is to (1) disable the capability to en-queue another task/action during the time another one is being process and (2) display a message that indicates that the action has been sent.

### good ways to test this pr

in conclusion to what i have said in the goal, try to see that after clicking on each `action` button (except the `Build`) `Loading` message is displayed. In addition to that, testing that every attempt to execute two action is not possible (only the first one is being processed, while the rest are ignored).

### scenarios that I have tested
- Click on `Delete` and then `Edit` => Should not get "Page not found"
- Click on `Cancel pending build` and then `Edit` => Should not render the edit page
- Click on `Build` and in the modal on `Build` again and then try to click on `Delete` => Should not delete page.
- Click on `Delete` and `No` in the confirm => Make sure you can click `Delete` again or `Edit`, `Cancel pending build`, `Build` and etc are working.

hopefully, that has given you a clear picture.

File: app/views/hosts/show.html.erb [hunk 3]
```diff
@@ -111,3 +111,13 @@
     </div>
   </div>
 </div>
+<div id="processing_message" class="modal fade hide">
+  <div class="modal-dialog modal-sm">
+    <div class="modal-content">
+      <div class="modal-header">
+        <h4 class="modal-title"><%= _('Loading...') %></h4>
+      </div>
+      <div class="modal-body"><%= _('Please wait, while your request is being processed') %></div>
```
Comment (dLobatog) [inline app/views/hosts/show.html.erb:121]: Maybe use a spinner...? It strikes me as too verbose, maybe @ohadlevy @lzap @tbrisker have different opinions.

**Comment by dLobatog** [comment]
@ripcurld00d I found an issue with [foreman_bootdisk](http://github.com/theforeman/foreman_bootdisk). With this PR, clicking on the submenus of boot disk becomes impossible as the modal loads.

File: app/views/hosts/show.html.erb (hunk 3 above)
Comment (boaz0) [inline app/views/hosts/show.html.erb:121]: sure, we can do that.

**Comment by boaz0** [comment]
I have added the `spinner`, disabled closing the `Modal` by clicking on the background with the mouse or pressing `escape` key and set only the `Delete`, `Cancel pending build` and `Edit` buttons to display the `Modal` (processing message).

IMHO, displaying a process message when clicking on `Boot disk` should be handled in the plugin to avoid breaking the UI.

I have no idea, if I should display that message when `Run Puppet` is clicked, though.

**Comment by boaz0** [comment]
[test]

**Comment by dLobatog** [comment]
@ripcurld00d This is looking good enough, thanks! Merging..

**Comment by dLobatog** [comment]
Merged as c4698fe9313e8d292373fa5fabb5bf76fd8897d3, thanks @ripcurld00d !
